python/ray/serve/deployment_state.py

```diff
@@ -833,33 +816,23 @@ def deploy(self, deployment_info: DeploymentInfo) -> Tuple[Optional[GoalId], boo
                 and deployment_info.version is not None
                 and existing_info.version == deployment_info.version
             ):
-                return self._curr_goal, False
-
-        # Keep a copy of previous deployment info in case goal failed to
-        # complete to initiate rollback.
-        self._rollback_info = self._target_info
+                return False
```

Comment:
The deployment status should be set to `RUNNING` before returning `False` since it gets set to `UPDATING` in `_set_deployment_goal()`.